kedro/framework/cli/pipeline.py

```diff
@@ -1064,28 +1063,38 @@ def _delete_artifacts(*artifacts: Path):
 
 
 def _append_package_reqs(
-    requirements_in: Path, package_reqs: List[str], pipeline_name: str
+    requirements_txt: Path, package_reqs: List[str], pipeline_name: str
 ) -> None:
-    """Appends modular pipeline requirements to project level requirements.in"""
-    existing_reqs = _safe_parse_requirements(requirements_in.read_text())
+    """Appends modular pipeline requirements to project level requirements.txt"""
     incoming_reqs = _safe_parse_requirements(package_reqs)
-    reqs_to_add = set(incoming_reqs) - set(existing_reqs)
-    if not reqs_to_add:
-        return
-
-    sorted_reqs = sorted(str(req) for req in reqs_to_add)
-    sep = "\n"
-    with open(requirements_in, "a", encoding="utf-8") as file:
-        file.write(
-            f"\n\n# Additional requirements from modular pipeline `{pipeline_name}`:\n"
+    if requirements_txt.is_file():
+        existing_reqs = _safe_parse_requirements(requirements_txt.read_text())
+        reqs_to_add = set(incoming_reqs) - set(existing_reqs)
+        if not reqs_to_add:
+            return
+
+        sorted_reqs = sorted(str(req) for req in reqs_to_add)
+        sep = "\n"
+        with open(requirements_txt, "a", encoding="utf-8") as file:
+            file.write(
+                f"\n\n# Additional requirements from modular pipeline `{pipeline_name}`:\n"
+            )
+            file.write(sep.join(sorted_reqs))
+        click.secho(
+            f"Added the following requirements from modular pipeline `{pipeline_name}` to "
+            f"requirements.txt:\n{sep.join(sorted_reqs)}"
         )
-        file.write(sep.join(sorted_reqs))
-    click.secho(
-        f"Added the following requirements from modular pipeline `{pipeline_name}` to "
-        f"requirements.in:\n{sep.join(sorted_reqs)}"
-    )
+    else:
```

Comment:
```suggestion
    elif incoming_reqs:
```
If the pipeline also doesn't have any requirements, this would otherwise create an empty file `requirements.txt`, which is not that useful.

Reply:
Actually I got confused by this block, as I see tests pass. Does this 

```python
with open(requirements_txt, "a", encoding="utf-8") as file:
    file.write(sep.join(sorted_reqs))
```

not create the file if there's nothing to append?

Reply:
There's a check further up the chain to check whether there are any requirements to append at all: 
https://github.com/quantumblacklabs/kedro/blob/develop/kedro/framework/cli/pipeline.py#L281